engine/src/main/java/com/google/android/fhir/search/MoreSearch.kt

```diff
@@ -93,20 +100,49 @@ fun Search.getQuery(isCount: Boolean = false): SearchQuery {
     }
   }
 
+  filterStatement += nestedSearches.nestedQuery(filterStatement, filterArgs, type)
+  val whereArgs = mutableListOf<Any>()
   val query =
-    """
-    SELECT ${ if (isCount) "COUNT(*)" else "a.serializedResource" }
-    FROM ResourceEntity a
-    $sortJoinStatement
-    WHERE a.resourceType = ?
-    $filterStatement
-    $sortOrderStatement
-    $limitStatement
-    """
+    when {
+        isCount -> {
+          """ 
+        SELECT COUNT(*)
```

Comment:
fix the indentation for here and for below

```suggestion
        """ 
        SELECT COUNT(*)
```

Reply:
Done